pkg/plugin/istio/install.go

```diff
@@ -496,45 +454,47 @@ func (p *IstioPlugin) installRemotes(remotePilotAddress string) error {
 func (p *IstioPlugin) clusterNetwork(cluster string) string {
 	c, err := p.KarmadaClient().ClusterV1alpha1().Clusters().Get(context.TODO(), cluster, metav1.GetOptions{})
 	if err != nil {
-		// fallback to generated network
-		return networkName(cluster)
+		// fallback to primary cluster's network
+		logrus.Warnf("cluster %s fallback to default network", cluster)
+		return defaultNetworkName
 	}
 
 	if n, ok := c.Labels[istiolabel.TopologyNetwork.Name]; ok {
 		return n
 	}
 
-	return networkName(cluster)
-}
+	if cluster == p.args.Primary {
+		logrus.Infof("primary cluster %s use default network", cluster)
+		return defaultNetworkName
+	}
 
-func networkName(cluster string) string {
-	logrus.Warnf("fallback to generate network name, cluster %s", cluster)
-	return fmt.Sprintf("%s-network", cluster)
+	logrus.Infof("cluster %s use primary's network", cluster)
+	// return primary cluster's network if not sepcifies
+	return p.clusterNetwork(p.args.Primary)
```

Comment:
I could not understand this.

Reply:
IMO, if we make default network empty, we donot need this, right?

Reply:
yes